utils/prometheus/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+extern crate lazy_static;
+
+use futures_util::{FutureExt, future::Future};
+use hyper::http::StatusCode;
+use hyper::{Server, Body, Request, Response, service::{service_fn, make_service_fn}};
+use prometheus::{Encoder, Opts, TextEncoder, core::Atomic};
+use std::net::SocketAddr;
+#[cfg(not(target_os = "unknown"))]
+mod networking;
+
+pub use prometheus::core::{
+	GenericGauge as Gauge, AtomicF64 as F64, AtomicI64 as I64, AtomicU64 as U64
+};
+pub use lazy_static::lazy_static;
+
+pub fn create_gauge<T: Atomic + 'static>(name: &str, description: &str) -> Gauge<T> {
+	let opts = Opts::new(name, description);
+	let gauge = Gauge::with_opts(opts).expect("Creating Gauge Failed");
+	prometheus::register(Box::new(gauge.clone())).expect("Registering gauge failed");
```

Comment:
This will use the default (global) registry. While convenient I think we should not use it.

Reply:
What do you suggest we should do instead?